src/coreComponents/constitutive/fluid/multifluid/compositional/CompositionalMultiphaseFluid.cpp

```diff
@@ -66,45 +62,61 @@ CompositionalMultiphaseFluid::CompositionalMultiphaseFluid( string const & name,
     setDescription( "Table of binary interaction coefficients" );
 }
 
-integer CompositionalMultiphaseFluid::getWaterPhaseIndex() const
+template< typename FLASH, typename PHASE1, typename PHASE2, typename PHASE3 >
+integer CompositionalMultiphaseFluid< FLASH, PHASE1, PHASE2, PHASE3 >::getWaterPhaseIndex() const
 {
-  string const expectedWaterPhaseNames[] = { "water" };
-  return PVTProps::PVTFunctionHelpers::findName( m_phaseNames, expectedWaterPhaseNames, viewKeyStruct::phaseNamesString() );
+  return -1;
```

Comment:
Is there a plan yet for how we handle the water phase index in the new design?

Reply:
Currently I'm not sure how the `phaseNames` input fits into all this. I haven't put this in since my current implementations are 2-phase with liquid and vapour. Once I start doing implementations with water, this will have to be worked out.